crowbar_framework/app/models/service_object.rb

```diff
@@ -1045,6 +1049,16 @@ def apply_role(role, inst, in_queue, bootstrap = false)
       end
     end
 
+    if skip_unchanged_nodes_enabled
```

Comment:
Hmm, I was thinking earlier than that ;-) When we check if the proposal has to be queued. Or is it not possible there? (Or maybe between what I suggest and what you did, it actually doesn't change things in reality?)

Reply:
Any reason to move it into the queue part? I dont see any tbh and introducing this into the part where we check if we should delay the proposal or not seems a bit of mixing logic. The `skip_unready_nodes` was ok to introduce there because it affected the queuing part but this does not seem like a good fit for that, unless you are thinking of expanding this to also cover the unready nodes as well?

Reply:
Or maybe you want to avoid loading the node objects for nodes we are not going to use anyway? In that case it would make sense to have it earlier that this, yeah...

Reply:
Umm, Im finding it difficult to manage it that way properly as we could have both options for skipping the nodes active at the same time, so I would have to add more complexity into the mix to manage `new_elements` properly.

Do we really want to add more complexity to `apply_role`? Indeed we could gain some time by not loading the nodes that we are not gonna apply from the chef server but im not sure if in terms of maintainability its worth it :(

Reply:
> Umm, Im finding it difficult to manage it that way properly as we could have both options for skipping the nodes active at the same time, so I would have to add more complexity into the mix to manage new_elements properly.

Let's chat tomorrow; I'm clearly missing something as I don't see the additional complexity.